include/exadg/incompressible_navier_stokes/postprocessor/postprocessor.h

```diff
@@ -73,17 +75,47 @@ class PostProcessor : public PostProcessorBase<dim, Number>
   setup(Operator const & pde_operator) override;
 
   void
-  do_postprocessing(VectorType const & velocity,
-                    VectorType const & pressure,
-                    double const       time             = 0.0,
-                    int const          time_step_number = numbers::steady_timestep) override;
+  do_postprocessing(VectorType const &     velocity,
+                    VectorType const &     pressure,
+                    double const           time             = 0.0,
+                    types::time_step const time_step_number = numbers::steady_timestep) override;
 
 protected:
   MPI_Comm const mpi_comm;
 
 private:
+  void
+  initialize_additional_vectors();
+
+  // TODO: FOR A MODULAR DESIGN THERE SHOULD PROBALY BE A CLASS CALLED MEAN_VECTOR_CALCULATION IN
+  // WHICH RELEVANT SAMPLES CAN JUST BE SUBMITTED.
+  void
+  compute_mean_velocity(VectorType & mean_velocity, VectorType const & velocity);
```

Comment:
I think there should be a calculator for this. But I don't want to do it in this PR since it is already quite big.

Reply:
Good idea. Definitely not in this PR, full agreement from my side.